test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestApiParser.java

```diff
@@ -42,7 +42,11 @@ public ClientYamlSuiteRestApi parse(String location, XContentParser parser) thro
                 if ("methods".equals(parser.currentName())) {
                     parser.nextToken();
                     while (parser.nextToken() == XContentParser.Token.VALUE_STRING) {
-                        restApi.addMethod(parser.text());
+                        String method = parser.text();
+                        if (restApi.getMethods().contains(method)) {
+                            throw new IOException("Found duplicate method [" + method + "]");
```

Comment:
I used the same exception type that was already used and it did not feel weird to me. We could use `ElasticsearchParseException` instead, would that be more adapted to you?